app/src/main/java/com/andrew/tasky/agenda/data/event/EventRepositoryImpl.kt

```diff
@@ -26,112 +26,145 @@ class EventRepositoryImpl @Inject constructor(
-                            filename = eventPhoto.key,
-                            body = compressedImageByte.toRequestBody()
+                            name = "update_event_request",
+                            value = Json.encodeToString(event.toUpdateEventRequest())
+                        ),
+                    photoData = event.photos.filterIsInstance<EventPhoto.Local>().mapIndexed {
+                        index, eventPhoto ->
+                        val uriByteConverter = UriByteConverter(
+                            appContext = appContext,
+                            ioDispatcher = ioDispatcher
+                        )
+                        val imageByte = uriByteConverter.uriToByteArray(
+                            uri = Uri.parse(eventPhoto.uri),
+                        )
+                        val compressedImageByte = BitmapCompressor().compressByteArray(
+                            byteArray = imageByte,
+                            targetSize = 1000000
                         )
+                        MultipartBody.Part
+                            .createFormData(
+                                name = "photo$index",
+                                filename = eventPhoto.key,
+                                body = compressedImageByte.toRequestBody()
+                            )
+                    }
+                )
+            }
+            return when (result) {
+                is AuthResult.Authorized -> {
+                    result.data?.let {
+                        db.getEventDao().upsertEvent(
+                            it.toEventEntity(
+                                isDone = event.isDone,
+                                isGoing = event.isGoing
+                            )
+                        )
+                    }
+                    AuthResult.Authorized()
                 }
-            )
-        }
-        when (result) {
-            is AuthResult.Authorized -> {
-                result.data?.let {
-                    db.getEventDao().upsertEvent(
-                        it.toEventEntity(
-                            isDone = event.isDone,
-                            isGoing = event.isGoing
+                is AuthResult.Unauthorized -> {
+                    db.getEventDao().upsertModifiedEvent(
+                        ModifiedEventEntity(
+                            id = event.id,
+                            modifiedType = ModifiedType.UPDATE
+                        )
+                    )
+                    AuthResult.Unauthorized()
+                }
+                is AuthResult.UnknownError -> {
+                    db.getEventDao().upsertModifiedEvent(
+                        ModifiedEventEntity(
+                            id = event.id,
+                            modifiedType = ModifiedType.UPDATE
                         )
                     )
+                    AuthResult.Unauthorized()
                 }
```

Comment:
If multiple branches of a when expression contain the same code, you can combine them:
```
is AuthResult.Unauthorized,
is AuthResult.UnknownError -> {
    // ...
}
```